src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,82 @@
+import { FC } from 'react';
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (value: number) => void;
+};
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const amountOfPages = getNumbers(1, Math.ceil(total / perPage));
+
+  const handlePageChange = (
+    event: React.MouseEvent<HTMLLIElement, MouseEvent>,
+  ) => {
+    const element = event.target as HTMLElement;
+    const page = element.textContent;
+
+    onPageChange(Number(page));
+  };
```

Comment:
The click handler on the page number relies on the text content of the HTML element, which may not be robust. It's better to use a data attribute or another method to pass the page number explicitly.